### ISSUE TITLE
[suggestion] Comments

### ISSUE THREAD

**Issue by BernsteinA** [SpektroAudio/Patchbook#7]
I propose that comments be added to the spec.
Maybe `#` or `//` ?

**Comment by cristianvogel** [comment]
I second that, and suggest // or /* */  

It reflects when modules or reconfigurations are temporarily repatched or removed then the change is assessed but should be annotated

**Comment by cristianvogel** [comment]
Oh - did you mean comments that the parser would ignore in the script or comments that will be rendered into the graph? 

I meant comments that would be ignored.. But, I think it would be very valuable to have a user comment field that would appear under the paramater settings, to annotate gesture/performance changes.
 
Example:
* Braids:
| Mode = Fold
| Timbre = 30%
| Timbre CV = -20%
| Color = 0%
"Turn color up for finale"

**Comment by robgmsn** [comment]
"Turn color up for finale" :D

Any of the lines which follow a | will appear below the module, so you could do...

* Braids:
| Mode = Fold
| Timbre = 30%
| Timbre CV = -20%
| Color = 0%
| Turn color up for finale

I'm just looking into the adding comments which the parser would ignore...

**Comment by cristianvogel** [comment]
Hehe yes full colour blast at finale!

What I thought is actually that it could be styled differently to the parameter list... italics, smaller font...

As it could get a bit long and mess up the layout - such annotation would need to be less prominent than the parameter settings I guess

🔝

**Comment by robgmsn** [comment]
Yeah, that makes sense.

I've just added the ability to use //comments and submitted a pull request, but I think the graphviz comments need a change to the data model.

**Comment by icaroferre** [comment]
Hey guys, I think // (for single lines) and /* */ (for multiple lines) should work fine. 
@robgmsn I've seen your pull request but it looks like you just added conditional to ignore those but I think I'm actually create a regex filter to pull them out and store them somewhere in the data structure since it's probably a safer approach.

**Comment by icaroferre** [comment]
Comments like "Turn color up for finale" can just be written as Color = 100% however I can think of some comments that would make more sense as comments rather than parameters (like "Change mode to FOLD during chorus").

I think I'm gonna make it so comments that start with //  followed by module name and = (like parameters) are added to the module itself, otherwise they're included as global comments to the patch.

**Comment by robgmsn** [comment]
Yeah, sounds good. Two birds with one stone 👍

**Comment by BernsteinA** [comment]
I was thinking about comments that are totally ignored by the parser.
For stuff that ends up in the patch, maybe a better way to think of it would be annotations, rather than comments?

**Comment by icaroferre** [comment]
@BernsteinA Anything that is not formatted as a voice, connection, or parameter should be ignored by the parser.

**Comment by BernsteinA** [comment]
Personally I like putting comments at the end of lines, which afaict would currently just get included by the regex as part of the parameter value
```
* Braids:
| Mode = Fold
| Timbre = 30% // TODO: fine tune this
| Timbre CV = -20%
| Color = 0%
```

Silently ignoring lines that don't match anything is ok, but maybe it would be good to carve out a reserved character for starting comment lines with?

**Comment by icaroferre** [comment]
I'm updating the parser right now to make the regex rules more precise so you should be able to add comments by just adding a breakline:

```
* Braids:  
| Mode = Fold  
| Timbre = 30%   
TODO: fine tune this  
| Timbre CV = -20%    
| Color = 0%  
```

The reason I don't wanna do annotations vs comments is that these 2 are quite similar and having both would add complexity for a very small trade-off.
I'd rather have just comments and change the graphviz method to optionally output comments or not.

**Comment by icaroferre** [comment]
Alright guys, I've added comments: https://github.com/SpektroAudio/Patchbook/commit/89ffe26b3a5ac3270daeef9b3e6edf01dbced020

So far I've only added single-line // comments but I'll add multi-line support asap.
All comments are global, at least for now. 
I do want to add specific module comments however I'm still thinking how the best approach would be display them (showing them below the parameter list could end up making the module node super long which would look silly).
A workaround would be to add the module comment as a parameter (as previously suggested) or to start the comment with "Module:" because that's what I'll probably use in the future to associate comments to specific modules.

Comments are also included in the GraphViz graph as a separate node.

**Comment by robgmsn** [comment]
Nice work man! This is a true great little project you've got here 💯

**Comment by cristianvogel** [comment]
Yes, annotation is such an important aspect of music it is so full of complex information. Well done for initiating this project for modular synthesiser practice, as we have no UNDO or any way to quickly recall configurations
